### DISCUSSION TITLE
RFC: Passing user preferences to blocks

### DISCUSSION THREAD

**Discussion by davidebriani** [blockprotocol/blockprotocol#315] [RFC]
## Goals

Provide blocks with user preferences that would help them to be visually consistent with the application embedding them. For instance:
- Language of choice
- Currency of choice
- Date formatting

## Proposal

The embedding app could specify a locale of choice for the user, formatted as per [RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646) (e.g. en-US)

**Comment by vilkinsons** [comment]
Hi Davide, thanks for your suggestion. These particular examples will be covered by planned internationalization (i18n) support, which we still need to write a bit about. That work could be extended to encapsulate other user preferences and context variables.
